### PR TITLE
Play 3.0

### PR DESCRIPTION
**Pull request by AshCorr**
## What does this change

Bumps Play to 3.0. Akka has been replaced with Pekko in this version of Play, it should be a drop in replacement for the most part but I've had to update some import paths to use the new package name!

Probably wants https://github.com/leanovate/play-mockws/pull/321 to be merged & released first! This PR includes many commits from #321

We're using this library in https://github.com/guardian/frontend and [would like to switch to Play 3.0](https://github.com/guardian/frontend/pull/26677)!

### CODE REVIEW

**Comment by gaeljw** [comment]
Thanks for the work, I was about to suggest it or do it, and you already did :+1:

File: build.sbt [hunk 1]
```diff
@@ -4,22 +4,20 @@ scalacOptions ++= Seq("-deprecation", "-feature")
...
-
 libraryDependencies ++= Seq(
-  "com.typesafe.play"      %% "play"                    % playVersion % "provided",
-  "com.typesafe.play"      %% "play-ahc-ws"             % playVersion % "provided",
-  "com.typesafe.play"      %% "play-test"               % playVersion % "provided",
-  "org.scala-lang.modules" %% "scala-collection-compat" % "2.1.6"
+  "org.playframework"      %% "play"                    % playVersion % "provided",
+  "org.playframework"      %% "play-ahc-ws"             % playVersion % "provided",
+  "org.playframework"      %% "play-test"               % playVersion % "provided",
+  "org.scala-lang.modules" %% "scala-collection-compat" % "2.5.0"
```
Comment (mkurz) [inline build.sbt:15]: Now that you dropped Scala 2.12, this library is not needed anymore.
See the project's [README](https://github.com/scala/scala-collection-compat):
> This library makes some Scala 2.13 APIs available on Scala 2.11 and 2.12.

 Scala 2.13+ has this built-in now, you can replace code like:
```scala
import scala.collection.JavaConverters._
```
with
```scala
import scala.jdk.CollectionConverters._
```

@rtyley did the same in https://github.com/guardian/play-secret-rotation/pull/411

File: README.md [hunk 2]
```diff
@@ -43,6 +43,10 @@ await(ws.url("http://dns/url").get()).body == "http response"
 
 Add MockWS as test dependency in the `build.sbt`:
 
+* for Play 2.9.x:
```
Comment (mkurz) [inline README.md:46]: Shouldn't this be Play 3.0.x based on the things you change in this pull request? (Switching to pekko, setting `val playVersion = "3.0.0"` in `build.sbt`)

File: README.md [hunk 3]
```diff
@@ -143,6 +147,7 @@ Other examples can be found in the [tests](src/test/scala/mockws/).
 
 ## Compatibility Matrix
 
+- MockWS 2.9.x is actually only compatible with Play 2.9.y., with Scala 3 or 2.13.
```
Comment (mkurz) [inline README.md:150]: Same here, Play 3.0.x? Or not?

File: README.md (hunk 2 above)
Comment (mkurz) [reply README.md:46]: Ah sorry, I totally missed #321, anyway you probably want to add entires for Play 3.0.0 then in the README :wink: Sorry for the noise

File: build.sbt (hunk 1 above)
Comment (AshCorr) [reply build.sbt]: Good catch, updated!

**Comment by gaeljw** [comment]
Do you know if there's a chance this PR gets merged and released ? I hadn't noticed but the project seems unmaintained according to https://github.com/leanovate/play-mockws/issues/268 or maybe maintainers changed since then?

I mean no offense to anyone but just wondering if we'll have to consider forking the project to get the benefit of this PR or if we can expect a new release of it.

**Comment by AshCorr** [comment]
> Do you know if there's a chance this PR gets merged and released ?

Not sure unfortunately, I don't have any contact with the maintainers of this project.

I did notice @avdv gave this PR a 👍 and they've previously merged PRs on this repo, maybe they're able to do something?

**Review by adriana-miclos-depop — APPROVED** [review]
LGTM!

**Review by gaeljw — APPROVED** [review]

**Comment by avdv** [comment]
Hi there!

@AshCorr  thank you for helping out to keep up with the latest Play releases!

I'll have a look over the next days, I have no idea what I need to do to put out a release... :smile_cat:

**Comment by AshCorr** [comment]
> Hi there!
> 
> @AshCorr thank you for helping out to keep up with the latest Play releases!
> 
> I'll have a look over the next days, I have no idea what I need to do to put out a release... 😸

Thank you, and sorry to put you on the spot 😅 Feel free to DM on Discord if you need any help `@thisisanameilike`!

**Comment by gaeljw** [comment]
Just for everyone to know, I aim to set some more automation for the project in the coming days, and then we'll merge and release this :)

We might also take the opportunity of the major version to change the way we distribute it and implement https://github.com/leanovate/play-mockws/issues/329. If anyone has any feelings about this, please shout out :)

**Comment by gaeljw** [comment]
@AshCorr I'm going to rebase your work on https://github.com/leanovate/play-mockws/pull/335

**Comment by gaeljw** [comment]
@AshCorr I allowed myself to rebase your work on top of https://github.com/leanovate/play-mockws/pull/335

**Pull request merged**